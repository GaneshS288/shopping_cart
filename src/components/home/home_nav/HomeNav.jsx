import { NavLink } from "react-router-dom";

function HomeNav() {
    return(
        <>
        <h2>Categories</h2>
        <nav>
            <button>
                <NavLink to={"/home/products"}>All</NavLink>

            </button>

            <button>
                <NavLink to={"/home/electronics"}>Electronics</NavLink>
            </button>

            <button>
                <NavLink to={"/home/jewelery"}>Jewelery</NavLink>
            </button>

            <button>
                <NavLink to={"/home/men's clothing"}>Men&apos;s clothing</NavLink>
            </button>

            <button>
                <NavLink to={"/home/women's clothing"}>women&apos;s clothing</NavLink>
            </button>
        </nav>
        </>
    );
}

export default HomeNav;