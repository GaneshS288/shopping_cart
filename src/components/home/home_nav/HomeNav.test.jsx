import { describe, it, expect, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import { createMemoryRouter, RouterProvider } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import HomeNav from "./HomeNav";

function testSetup() {
  const routes = [
    {
      path: "/home",
      element: <HomeNav></HomeNav>,
    },
  ];

  const router = createMemoryRouter(routes, {
    initialEntries: ["/", "/home"],
    initialIndex: 1,
  });

  const user = userEvent.setup();

  return { router, user };
}

describe("Home Nav", () => {
  it("matches snapshot", () => {
    const { router } = testSetup();
    const { container } = render(
      <RouterProvider router={router}></RouterProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it("renders 5 links", () => {
    const { router } = testSetup();
    render(<RouterProvider router={router}></RouterProvider>);

    expect(screen.getAllByRole("link").length).toBe(5);
  });
});
