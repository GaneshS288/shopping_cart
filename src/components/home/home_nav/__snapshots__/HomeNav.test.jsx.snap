// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Nav > matches snapshot 1`] = `
<div>
  <h2>
    Categories
  </h2>
  <nav>
    <button>
      <a
        class=""
        data-discover="true"
        href="/home/products"
      >
        All
      </a>
    </button>
    <button>
      <a
        class=""
        data-discover="true"
        href="/home/electronics"
      >
        Electronics
      </a>
    </button>
    <button>
      <a
        class=""
        data-discover="true"
        href="/home/jewelery"
      >
        Jewelery
      </a>
    </button>
    <button>
      <a
        class=""
        data-discover="true"
        href="/home/men's clothing"
      >
        Men's clothing
      </a>
    </button>
    <button>
      <a
        class=""
        data-discover="true"
        href="/home/women's clothing"
      >
        women's clothing
      </a>
    </button>
  </nav>
</div>
`;
